designs/replicated-pv/mayastor/pool-cordon.md

```diff
@@ -106,19 +112,67 @@ As such we're proposing that the CR reflects only the cordon state of the pool,
 
 #### Internal gRPC
 
-TODO
+Add cordon handlers to the [PoolGrpc](https://github.com/openebs/mayastor-control-plane/blob/2fee47f970bc73c4e5ddb83a1b390d5ab7074b4b/control-plane/grpc/proto/v1/pool/pool.proto#L175).
+Example:
+
+```protobuf
+service PoolGrpc {
+    rpc CordonPool (CordonPoolRequest) returns (CordonPoolReply) {}
+    rpc UncordonPool (UncordonPoolRequest) returns (UncordonPoolReply) {}
+}
+
+message CordonPoolRequest {
+  string node_id = 1;
+  string pool_id = 2;
+  string label = 3;
+}
+
+message CordonPoolReply {
+  oneof reply {
+    Node node = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+message UncordonPoolRequest {
+  string node_id = 1;
+  string pool_id = 2;
+  string label = 3;
+}
+
+message UncordonPoolReply {
+  oneof reply {
+    Node node = 1;
+    common.ReplyError error = 2;
+  }
+}
+```
 
 #### Public OpenAPI
 
-TODO
+```openapi
+'/nodes/{node_id}/pools/{pool_id}/cordon/{label}':
+    put:
+        tags:
+        - Pools
+        operationId: put_node_pool_cordon
+    delete:
+        tags:
+        - Pools
+        operationId: put_node_pool_cordon
+```
 
 #### Kubectl Plugin
 
-TODO
+Add new subcommands to cordon with the pool resource.
 
 ### Test Plan
 
-TODO
+In order to test this feature we need a scenario where we'd make use the pool which we are cordoning.
+If there are multiple options available, what if the scheduler always picks the other pools and we never exercise the cordoning logic for example.
```

Comment:
What does this mean?

Reply:
I see, in that case we probably need to have exactly identical pools so that the other factors are ruled out? Not sure how simple it can be.